Source/WebCore/Modules/indexeddb/IDBRequest.h

```diff
@@ -196,9 +198,10 @@ class IDBRequest : public EventTarget, public IDBActiveDOMObject, public ThreadS
     IndexedDB::IndexRecordType m_requestedIndexRecordType { IndexedDB::IndexRecordType::Key };
 
     bool m_shouldExposeTransactionToDOM { true };
-    bool m_hasPendingActivity { true };
     bool m_hasUncaughtException { false };
     RefPtr<Event> m_eventBeingDispatched;
+    enum class PendingActivityType : uint8_t { EndingEvent, CursorIteration, None };
+    std::atomic<PendingActivityType> m_pendingActivity { PendingActivityType::EndingEvent };
```

Comment:
Why does this need to be atomic? It is usually not needed for pending activity logic so this is potentially incorrect.

Reply:
Would be nice to move it up with the booleans for better packing.

Reply:
IDBRequest can be created and accessed on worker thread (see ASSERT in `IDBRequest::virtualHasPendingActivity()`), and I am not sure whether worker thread is paused when `hasPendingActivity` is called... so I use atomic. Is worker thread also paused?

Reply:
Ok! I will remove atomic before landing